src/step-function.ts

```diff
@@ -528,23 +555,40 @@ abstract class BaseStepFunction<
 
     const { name, input, traceHeader } = retrieveMachineArgs(call);
 
-    return {
-      Type: "Task" as const,
-      Resource: `arn:aws:states:::aws-sdk:sfn:${
-        this.resource.stateMachineType ===
-        aws_stepfunctions.StateMachineType.EXPRESS
-          ? "startSyncExecution"
-          : "startExecution"
-      }`,
-      Parameters: {
-        StateMachineArn: this.resource.stateMachineArn,
-        ...(input ? context.toJsonAssignment("Input", input) : {}),
-        ...(name ? context.toJsonAssignment("Name", name) : {}),
-        ...(traceHeader
-          ? context.toJsonAssignment("TraceHeader", traceHeader)
-          : {}),
-      },
+    const inputs = {
+      Input: input,
+      Name: name,
+      TraceHeader: traceHeader,
     };
+
+    return context.evalContext(call, (evalExpr) => {
+      // evaluate each of the input expressions,
+      // returning an object assignment with the output value { input.$: $.inputLocation }
+      // and a state object containing the output and/or a sub-state with additional required nodes to add to the
+      // machine
+      const evalInputs = Object.entries(inputs)
+        .filter(([, expr]) => !!expr)
+        .map(([key, expr]) => context.toJsonAssignment(key, evalExpr(expr!)));
```

Comment:
Should this be done as part of the interpreter instead of in the integration?

Reply:
You mean like wrap each of the states in new types and do some of the leaky transformation later?

```ts
const state = ASLGraph.Pass = {
   Type: "Task",
   parameters: {
       StateMachineArn: this.resource.stateMachineArn,
       Input: input ? evalExpr(input) : undefined,
       Name: name ? evalExpr(name) : undefined,
       TraceHeader: traceHeader ? evalExpr(traceHeader) : undefined,
   }
}
```

later...


```ts
const task = ASLGraph.graphTaskToTask(graphTask);
```

```ts
function graphTaskToTask(task: ASLGraph.Task): Task {
   return {
      Type: "Task",
      ...,
      Parameters: // ... determine if the values need `.$` or not
   }
}
```